README
NAME

    Dist::Zilla::Plugin::Git::FilePermissions - fix the file permissions in
    your Git repository with Dist::Zilla

VERSION

    Version 1.001

SYNOPSIS

      # in dist.ini:
      [Git::FilePermissions]
      perms = ^bin/         0755
      perms = ^scripts/     0755

DESCRIPTION

    This plugin fixes the file permissions of all the files in the Git
    repository where your project is saved. Files not in the Git index, and
    directories, are ignored.

    Without configuration, every file is changed to the default permission
    of 0644. The default permissions can be changed with the default option
    and you can configure different permissions for some files with the
    perms option in the dist.ini.

    The plugin runs in the before build phase, which means it will fix the
    file permissions before the files are picked up in the file gather
    phase. The new permissions are therefore also the ones used in the
    build.

    The plugin should ensure that you always commit your files with the
    correct permissions.

 perms

    The perms configuration option takes the form of:

      perms = REGEX WHITESPACE PERMS

    or

      perms = REGEX WHITESPACE -

    The perms configuration options are processed in order for every file.
    If a file matches the REGEX the file permissions are changed to the
    corresponding PERMS instead of the default permissions of 0644. If the
    PERMS are - the file is ignored.

SUPPORT

 Bugs / Feature Requests

    Please report any bugs or feature requests through the issue tracker at
    https://github.com/skirmess/Dist-Zilla-Plugin-Git-FilePermissions/issues.
    You will be notified automatically of any progress on your issue.

 Source Code

    This is open source software. The code repository is available for
    public review and contribution under the terms of the license.

    https://github.com/skirmess/Dist-Zilla-Plugin-Git-FilePermissions

      git clone https://github.com/skirmess/Dist-Zilla-Plugin-Git-FilePermissions.git

AUTHOR

    Sven Kirmess <sven.kirmess@kzone.ch>

SEE ALSO

    Dist::Zilla::Plugin::Git::RequireUnixEOL



Changes
Revision history for Dist-Zilla-Plugin-Git-FilePermissions

{{$NEXT}}

1.002  2022-10-25 20:34:22 UTC

  - Switch license to the ISC license
  - Use 'protocol.file.allow=always' for Git submodule tests, if required.

1.001  2022-01-08 22:42:13 UTC

  - Replace Git::Wrapper with Git::Background
  - Removed minimal dependency on Test::TempDir::Tiny 0.017 and Moose 2.1203
  - skip needs $how_many with Test::More < 1.302013_009
  - Removed autodie in tests
  - Code cleanup - moved all 'use' before 'has'

1.000  2018-04-02 22:38:46 UTC

  - No code changes
  - Distribution cleanup
  - Bumped dependency on Moose to 2.1203

0.004  2018-02-11 18:47:41 UTC

  - Ignore files that don't exist or where Git thinks they are a file but
    they are something else (e.g. submodules)

0.003  2017-10-20 23:39:10 UTC

  - Added a default option to specify default permissions

0.002  2017-10-16 21:46:46 UTC

  - Fix the tests for Windows

0.001  2017-10-15 19:39:17 UTC

  - Initial release


Makefile.PL
use 5.006;
use strict;
use warnings;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
    ABSTRACT           => 'fix the file permissions in your Git repository with Dist::Zilla',
    AUTHOR             => 'Sven Kirmess <sven.kirmess@kzone.ch>',
    CONFIGURE_REQUIRES => {
        'ExtUtils::MakeMaker' => 0,
    },
    DISTNAME         => 'Dist-Zilla-Plugin-Git-FilePermissions',
    LICENSE          => 'open_source',
    MIN_PERL_VERSION => '5.006',
    NAME             => 'Dist::Zilla::Plugin::Git::FilePermissions',
    PREREQ_PM        => {
        'Dist::Zilla::Role::BeforeBuild' => 0,
        'Git::Background'                => '0.003',
        Moose                            => 0,
        'Path::Tiny'                     => 0,
        'namespace::autoclean'           => 0,
        strict                           => 0,
        warnings                         => 0,
    },
    TEST_REQUIRES => {
        Carp                                  => 0,
        Cwd                                   => 0,
        'Dist::Zilla::Plugin::Git::GatherDir' => 0,
        Exporter                              => 0,
        'File::Basename'                      => 0,
        'File::Path'                          => '2.07',
        'File::Spec'                          => 0,
        'File::Spec::Functions'               => 0,
        'Test::DZil'                          => 0,
        'Test::Fatal'                         => 0,
        'Test::More'                          => '0.88',
        lib                                   => 0,
    },
    VERSION => '1.000',
    test    => {
        TESTS => 't/*.t',
    },
);

my %FallbackPrereqs = (
    Carp                                  => 0,
    Cwd                                   => 0,
    'Dist::Zilla::Plugin::Git::GatherDir' => 0,
    'Dist::Zilla::Role::BeforeBuild'      => 0,
    Exporter                              => 0,
    'File::Basename'                      => 0,
    'File::Path'                          => '2.07',
    'File::Spec'                          => 0,
    'File::Spec::Functions'               => 0,
    'Git::Background'                     => '0.003',
    Moose                                 => 0,
    'Path::Tiny'                          => 0,
    'Test::DZil'                          => 0,
    'Test::Fatal'                         => 0,
    'Test::More'                          => '0.88',
    lib                                   => 0,
    'namespace::autoclean'                => 0,
    strict                                => 0,
    warnings                              => 0,
);

if ( !eval { ExtUtils::MakeMaker->VERSION('6.63_03') } ) {
    delete $WriteMakefileArgs{TEST_REQUIRES};
    delete $WriteMakefileArgs{BUILD_REQUIRES};
    $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

if ( !eval { ExtUtils::MakeMaker->VERSION(6.52) } ) {
    delete $WriteMakefileArgs{CONFIGURE_REQUIRES};
}

WriteMakefile(%WriteMakefileArgs);

# vim: ts=4 sts=4 sw=4 et: syntax=perl
